src/NServiceBus.Core/InitializableEndpoint.cs

```diff
@@ -47,7 +49,15 @@ public InitializableEndpoint(SettingsHolder settings, IContainer container, List
             var transportInfrastructure = transportDefinition.Initialize(settings, connectionString);
             settings.Set<TransportInfrastructure>(transportInfrastructure);
 
-            var featureStats = featureActivator.SetupFeatures(container, pipelineSettings);
+            // should we use GetOrDefault<T>() ?? new T() instead to prevent "leaking" into the settings?
+            var routing = new RoutingComponent(
```

Comment:
Couldn't we just new these up internally in the component itself so that the component defines the lifecycle?